src/reset-password/ResetPasswordPage.jsx

```diff
@@ -78,22 +80,31 @@ const ResetPasswordPage = (props) => {
       props.validateToken(token);
       return <Spinner />;
     }
-  } else if (props.token_status === 'invalid') {
+  } else if (props.token_status === 'invalid' && props.errorCode !== INTERNAL_SERVER_ERROR) {
     return <InvalidTokenMessage />;
+  } else if (props.token_status === 'invalid' && props.errorCode === INTERNAL_SERVER_ERROR) {
+    return (
+      <div className="d-flex justify-content-center m-4">
+        <div className="d-flex flex-column mw-500">
+          <APIFailureMessage header={intl.formatMessage(messages['reset.password.token.validation.sever.error'])} />
+        </div>
+      </div>
+    );
```

Comment:
Move this condition after the below one so that you don't need to check for `props.errorCode === INTERNAL_SERVER_ERROR` in it.
```suggestion
  } else if (props.token_status === 'invalid' && props.errorCode === INTERNAL_SERVER_ERROR) {
    return (
      <div className="d-flex justify-content-center m-4">
        <div className="d-flex flex-column mw-500">
          <APIFailureMessage header={intl.formatMessage(messages['reset.password.token.validation.sever.error'])} />
        </div>
      </div>
    );
  } else if (props.token_status === 'invalid') {
    return <InvalidTokenMessage />;
```